Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc

```diff
@@ -122,42 +133,60 @@ void GEMGeometryBuilderFromDDD::build(GEMGeometry& theGeometry,
 
     doSuper = fv.nextSibling();
 
-    if (!loopExecuted)
+    if (!loopExecuted) {
       delete gemChamber;
+    }
   }
 
   // construct the regions, stations and rings.
   for (int re = -1; re <= 1; re = re + 2) {
     GEMRegion* region = new GEMRegion(re);
-    for (int st = 1; st <= GEMDetId::maxStationId; ++st) {
+    for (int st = 0; st <= GEMDetId::maxStationId; ++st) {
       GEMStation* station = new GEMStation(re, st);
       std::string sign(re == -1 ? "-" : "");
-      std::string name("GE" + sign + std::to_string(st) + "/1");
+      std::string name = (st == 0) ? "GE0" : ("GE" + sign + std::to_string(st) + "/1");
       station->setName(name);
+      bool foundSuperChamber = false;
       for (int ri = 1; ri <= 1; ++ri) {
         GEMRing* ring = new GEMRing(re, st, ri);
         for (auto superChamber : superChambers) {
           const GEMDetId detId(superChamber->id());
           if (detId.region() != re || detId.station() != st || detId.ring() != ri)
             continue;
 
-          superChamber->add(
-              theGeometry.chamber(GEMDetId(detId.region(), detId.ring(), detId.station(), 1, detId.chamber(), 0)));
-          superChamber->add(
-              theGeometry.chamber(GEMDetId(detId.region(), detId.ring(), detId.station(), 2, detId.chamber(), 0)));
+          foundSuperChamber = true;
+          int nstation = (st == 0) ? 6 : 2;
```

Comment:
magic numbers should be named constants

Reply:
@watson-ij Please use these constants. https://github.com/cms-sw/cmssw/blob/master/DataFormats/MuonDetId/interface/GEMDetId.h#L30-L31. I also suggest to rename `nstation` to `nlayers`.

Reply:
```suggestion
          int nlayers = (st == 0) ? GEMDetId::maxLayerId0 : GEMDetId::maxLayerId;
```

So, just to check @kpedro88 something like this would be okay, or I also need to introduce a named constant somewhere for `st==0`?

Reply:
You could use `GEMDetId::minStationId0`